spring/boot2-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaServerHttpResponse.java

```diff
@@ -287,7 +287,18 @@ public HttpHeaders getHeaders() {
         final HttpResponse response = HttpResponse.of(buildResponseHeaders());
         future.complete(response);
         logger.debug("{} Response future has been completed with an HttpResponse", ctx);
-        return Mono.fromFuture(response.whenComplete());
+
+        final CompletableFuture<Void> monoFuture = new CompletableFuture<>();
+        response.whenComplete().handle((unused, cause1) -> {
+            if (cause1 != null && !(cause1 instanceof CancelledSubscriptionException)) {
+                monoFuture.completeExceptionally(cause1);
+            } else {
+                // ignore CancelledSubscriptionException created when handling HTTP HEAD request
```

Comment:
Don't we need to check `ctx.request().method() == HttpMethod.HEAD`?

Reply:
I will check `HEAD` too!

Reply:
Had chat with @minwoox and I realized that HEAD is not to be checked here. 
@eunchan-kim Sorry for the wrong direction. Could you remove `ctx.request().method() == HttpMethod.HEAD` condition. 🙇‍♂️

Reply:
No problem. 
BTW, could you share the backgrounds of 
> I realized that HEAD is not to be checked here.
?

Reply:
The meaning of `cancel` signal is used when a `Subscriber` wants to clean up resources and it does not need data anymore. The publisher doesn't propagate a `CancelledSubscriptionException` to `Subscribe.onError(Throwable)`. `CancelledSubscriptionException` is used internally to indicate that a stream has been cancelled via `whenComplete()`.

https://github.com/line/armeria/blob/0bb591d252884ec471af00e2757032ae57196d5e/core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessage.java#L279-L282

So the `CancelledSubscriptionException` from the `whenComplete()` could be ignored.

Reply:
thanks for explanation! 👍